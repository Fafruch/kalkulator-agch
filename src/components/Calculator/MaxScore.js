import React from 'react'
import PropTypes from 'prop-types'

const MaxScore = ({ maxPrimaryScore, maxLingualScore }) => {
  let maxScore
  maxScore = maxPrimaryScore + maxLingualScore
  return (
    <div>
      <h3>
        <small> Tw√≥j wynik wynosi&nbsp;</small> <b> {maxScore} </b> <small>&nbsp;pkt.</small>
      </h3>
      <br />
    </div>
  )
}
MaxScore.propTypes = {
  maxPrimaryScore: PropTypes.number.isRequired,
  maxLingualScore: PropTypes.number.isRequired
}

export default MaxScore
