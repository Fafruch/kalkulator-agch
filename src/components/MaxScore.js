import React from 'react'
import PropTypes from 'prop-types'

const MaxScore = ({ maxPrimaryScoreWrapper, maxLingualScoreWrapper }) => {
  let maxScore
  let maxPrimaryScore
  let maxLingualScore

  if (maxPrimaryScoreWrapper === null) {
    maxPrimaryScore = 0
  } else {
    maxPrimaryScore = maxPrimaryScoreWrapper.computedScore
  }

  if (maxLingualScoreWrapper === null) {
    maxLingualScore = 0
  } else {
    maxLingualScore = maxLingualScoreWrapper.computedScore
  }

  maxScore = maxPrimaryScore + maxLingualScore

  return (
    <h5>
      Twój wynik wynosi <b>{maxScore}</b> punktów.
    </h5>
  )
}
MaxScore.propTypes = {
  maxPrimaryScoreWrapper: PropTypes.any.isRequired,
  maxLingualScoreWrapper: PropTypes.any.isRequired
}

export default MaxScore
