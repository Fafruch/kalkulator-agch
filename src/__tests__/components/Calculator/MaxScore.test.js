import React from 'react'
import { shallow } from 'enzyme'

import MaxScore from '../../../components/Calculator/MaxScore'

const setup = (setupProps = {}) => {
  const defaultProps = {
    maxPrimaryScore: 182,
    maxLingualScore: 192
  }
  const props = { ...defaultProps, ...setupProps }
  const wrapper = shallow(<MaxScore {...props} />)

  return {
    props,
    wrapper
  }
}

describe('MaxScore component', () => {
  test('should render', () => {
    const { wrapper } = setup()

    expect(wrapper).toMatchSnapshot()
  })

  test('should render total maxScore', () => {
    const { props, wrapper } = setup()

    expect(wrapper.find('b').text()).toEqual(` ${props.maxPrimaryScore * 4 + props.maxLingualScore} `)
  })
})
