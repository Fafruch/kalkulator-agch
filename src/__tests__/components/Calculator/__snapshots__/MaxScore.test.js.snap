// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MaxScore component should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MaxScore
    maxLingualScore={192}
    maxPrimaryScore={182}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          <small>
                     Twój maksymalny wynik wynosi 
          </small>
           
          <b>
                     
                    920
                     
          </b>
           
          <small>
                     pkt.
          </small>
</h3>,
        <br />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <small>
               Twój maksymalny wynik wynosi 
</small>,
            " ",
            <b>
               
              920
               
</b>,
            " ",
            <small>
               pkt.
</small>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": " Twój maksymalny wynik wynosi ",
            },
            "ref": null,
            "rendered": " Twój maksymalny wynik wynosi ",
            "type": "small",
          },
          " ",
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " ",
                920,
                " ",
              ],
            },
            "ref": null,
            "rendered": Array [
              " ",
              920,
              " ",
            ],
            "type": "b",
          },
          " ",
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": " pkt.",
            },
            "ref": null,
            "rendered": " pkt.",
            "type": "small",
          },
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            <small>
                         Twój maksymalny wynik wynosi 
            </small>
             
            <b>
                         
                        920
                         
            </b>
             
            <small>
                         pkt.
            </small>
</h3>,
          <br />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <small>
                 Twój maksymalny wynik wynosi 
</small>,
              " ",
              <b>
                 
                920
                 
</b>,
              " ",
              <small>
                 pkt.
</small>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": " Twój maksymalny wynik wynosi ",
              },
              "ref": null,
              "rendered": " Twój maksymalny wynik wynosi ",
              "type": "small",
            },
            " ",
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  920,
                  " ",
                ],
              },
              "ref": null,
              "rendered": Array [
                " ",
                920,
                " ",
              ],
              "type": "b",
            },
            " ",
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": " pkt.",
              },
              "ref": null,
              "rendered": " pkt.",
              "type": "small",
            },
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
